doc/users-guide/base/typeAliases.rst

```diff
@@ -0,0 +1,67 @@
+:title: Users Guide: Type Aliases
+
+.. _ug-typeAliases:
+
+Type Aliases
+============
+
+In addition to :ref:`declaring variables <ug-vars>` using ``var`` and
+:ref:`constants <ug-constParam>` using ``const`` and ``param``, Chapel
+supports the declaration of *type aliases* using the ``type`` keyword.
+As the term suggests, type aliases create symbolic names for types.
```

Comment:
I'm not so sure I like the exact 'typedef' proposal, but I *do* think we'd want different syntax for getting a "new type" and that it makes sense for the current usage to be an "alias type". We should probably move this discussion to some issue, but my counter-proposal would be:
```
 new type age = int;
```
or possibly
```
type age = new type int;
```

I like your reasoning about typedef/whatever not applying to argument intent/generic fields.